modules/tinymce/src/core/main/ts/html/Sanitization.ts

```diff
@@ -242,18 +242,34 @@ const sanitizeMathmlElement = (node: Element, settings: DomParserSettings) => {
   purify.addHook('uponSanitizeElement', (node, evt) => {
     // We know the node is an element as we have
     // passed an element to the purify.sanitize function below
-    const elm = node as Element;
-    const lcTagName = evt.tagName ?? elm.nodeName.toLowerCase();
+    const lcTagName = evt.tagName ?? node.nodeName.toLowerCase();
+    let keepElement;
 
     if (hasAllowedEncodings && lcTagName === 'semantics') {
       evt.allowedTags[lcTagName] = true;
+      keepElement = true;
+    } else if (lcTagName === 'annotation') {
+      keepElement = node.nodeType === NodeTypes.ELEMENT && hasValidEncoding(node as Element);
+    } else if (Type.isArray(settings.allow_mathml_elements)) {
+      keepElement = settings.allow_mathml_elements.includes(lcTagName);
+    } else {
+      return;
     }
 
-    if (lcTagName === 'annotation') {
-      const keepElement = hasValidEncoding(elm);
-      evt.allowedTags[lcTagName] = keepElement;
-      if (!keepElement) {
-        elm.remove();
+    evt.allowedTags[lcTagName] = keepElement;
+    if (!keepElement && settings.sanitize) {
+      if (node.nodeType === NodeTypes.ELEMENT) {
```

Comment:
You could use NodeType.isElement predicate then you don't need to cast node as element since that gets inferred by TS.